architecture-decision-records/0025-automatic-code-merge.md

```diff
@@ -0,0 +1,28 @@
+#25. Automatic code merging
+
+**Date:** 2022-03-15
+
+## Context
+We have two services now, dependabot and Scala Steward, which create pull requests for dependency updates. 
+We are getting around 5 or 6 of these per day which need to be merged in turn and can take a lot of time. We looked at using an automatic merging service.
+
+## Options considered
+We considered continuing merging the pull requests manually.
+
+The other choice was to use [Mergify](https://mergify.com/)
+
+## Considerations
+Because changes would be included in our service which a human hadn't seen before, we needed to make sure this was secure. 
+The service needed able to allow the following:
+* Restrict auto merging pull requests based on files changed. This way, we would only merge changes to dependency files.
+* Restrict auto merging until after all tests and security checks are complete.
+
+On top of this, the service should be well known and well established enough that we could be reasonably confident that it was secure and stable.
+
+## Decision
```

Comment:
Is it worth mentioning something around user access to Mergify, ie via GitHub, it thus being restricted?

Reply:
Yep, makes sense, I've added something in.